### PR TITLE
using scala.util.Try for URI.parse

### PR DESCRIPTION
**Pull request by 2is10**

### CODE REVIEW

File: shoebox/app/com/keepit/search/graph/URIGraph.scala [hunk 1]
```diff
@@ -205,7 +205,7 @@ class URIGraphImpl(indexDirectory: Directory, indexWriterConfig: IndexWriterConf
       }
 
       buildLineField(fieldName, domains, (fieldName, url) =>
-        URI.parse(url).flatMap{ uri =>
+        URI.parse(url).toOption.flatMap{ uri =>
```
Comment (andrewconner) [inline shoebox/app/com/keepit/search/graph/URIGraph.scala]: Do we need this to be an `Option`? `Try` has  `map()` and `flatMap()` as well.

Comment (andrewconner) [inline shoebox/app/com/keepit/search/graph/URIGraph.scala]: Same comment for all the `toOption` references.

**Comment by andrewconner** [comment]
Awesome! :)

File: shoebox/app/com/keepit/search/graph/URIGraph.scala (hunk 1 above)
Comment (2is10) [inline shoebox/app/com/keepit/search/graph/URIGraph.scala]: `Try.flatMap` requires a function that returns a `Try`, not an `Option`.

Comment (andrewconner) [inline shoebox/app/com/keepit/search/graph/URIGraph.scala]: Ah, good call.

**Pull request merged**